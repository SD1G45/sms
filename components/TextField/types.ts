import React from "react";

export interface TextFieldProps extends React.HTMLAttributes<HTMLInputElement> {
  value: string;
  label?: string;
  error?: boolean;
  errorMessage?: string;
  className?: string;
}
