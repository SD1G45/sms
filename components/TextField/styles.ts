import styled from "styled-components";
import { TextFieldProps } from "./types";

export const TextFieldContainer = styled.div`
  width: 100%;
`;

export const LabelContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Label = styled.label`
  display: block;
  padding-bottom: 7px;
`;

export const StyledLink = styled(Label)`
  color: ${(props) => props.theme.colors.primary};
  cursor: pointer;
`;

export const StyledTextField = styled.input<Omit<TextFieldProps, "label">>`
  padding: 12px;
  border: 1px solid
    ${(props) =>
      props.error ? props.theme.colors.error : props.theme.colors.lightGray};
  border-radius: 10px;
  width: 100%;
  margin-bottom: ${(props) => (props.error ? 8 : 0)}px;
`;

export const ErrorMessage = styled.p`
  color: ${(props) => props.theme.colors.error};
  font-size: 0.9rem;
  margin-left: 7px;
`;
