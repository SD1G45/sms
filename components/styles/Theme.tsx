import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
