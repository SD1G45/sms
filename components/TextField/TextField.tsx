import React from "react";
import { ErrorMessage, StyledTextField, TextFieldContainer } from "./styles";
import { TextFieldProps } from "./types";

const TextField: React.FC<TextFieldProps> = ({
  error,
  errorMessage,
  value,
  ...other
}) => {
  return (
    <TextFieldContainer>
      <StyledTextField type="text" error={error} {...other} value={value} />
      {error && errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
    </TextFieldContainer>
  );
};

export default TextField;
