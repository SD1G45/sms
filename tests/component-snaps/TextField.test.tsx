import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import TextField from "../../components/TextField";
import Theme from "../../components/styles/Theme";

test("TextField renders correctly", () => {
  const tree = renderer
    .create(
      <Theme>
        <TextField 
          value="TextField"
        />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("TextField with label renders correctly", () => {
  const tree = renderer
    .create(
      <Theme>
        <TextField 
          value="TextField"
          label="TextField"
        />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("TextField with error renders correctly", () => {
  const tree = renderer
    .create(
      <Theme>
        <TextField 
          value="TextField"
          error={true}
          errorMessage="There was an error"
        />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});