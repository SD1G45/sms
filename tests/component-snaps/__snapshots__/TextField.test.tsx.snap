// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: block;
  padding-bottom: 7px;
}

.c3 {
  color: #4881F0;
  cursor: pointer;
}

.c4 {
  padding: 12px;
  border: 1px solid #BEC8D2;
  border-radius: 10px;
  width: 100%;
  margin-bottom: 0px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      
    </label>
    <label
      className="c2 c3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      
    </label>
  </div>
  <input
    className="c4"
    value="TextField"
  />
</div>
`;

exports[`TextField with error renders correctly 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: block;
  padding-bottom: 7px;
}

.c3 {
  color: #4881F0;
  cursor: pointer;
}

.c4 {
  padding: 12px;
  border: 1px solid #FF7F7F;
  border-radius: 10px;
  width: 100%;
  margin-bottom: 8px;
}

.c5 {
  color: #FF7F7F;
  font-size: 0.9rem;
  margin-left: 7px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      
    </label>
    <label
      className="c2 c3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      
    </label>
  </div>
  <input
    className="c4"
    value="TextField"
  />
  <p
    className="c5"
  >
    There was an error
  </p>
</div>
`;

exports[`TextField with label renders correctly 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: block;
  padding-bottom: 7px;
}

.c3 {
  color: #4881F0;
  cursor: pointer;
}

.c4 {
  padding: 12px;
  border: 1px solid #BEC8D2;
  border-radius: 10px;
  width: 100%;
  margin-bottom: 0px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      TextField
    </label>
    <label
      className="c2 c3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      
    </label>
  </div>
  <input
    className="c4"
    value="TextField"
  />
</div>
`;
